pkg/apis/openstack/validation/controlplane_test.go

```diff
@@ -318,11 +320,53 @@ var _ = Describe("ControlPlaneConfig validation", func() {
 
 			controlPlane.LoadBalancerClasses = []api.LoadBalancerClass{lbClasses[1]}
 
-			errorList := ValidateControlPlaneConfigAgainstCloudProfile(controlPlane, differentRegion, cloudProfile, cloudProfileConfig, nilPath)
+			errorList := ValidateControlPlaneConfigAgainstCloudProfile(controlPlane, differentRegion, fpName, cloudProfile, cloudProfileConfig, nilPath)
+
+			Expect(errorList).To(BeEmpty())
+		})
+
+		It("should pass for a load balancer class which is assigned by a wildcard FloatingPoolName", func() {
+			fpName := "some_name"
+			lbClasses := []api.LoadBalancerClass{
+				{Name: "lbClass1"},
+				{Name: "lbClass2"},
+				{Name: "lbClass2"},
+			}
+
+			cloudProfileConfig.Constraints.FloatingPools = append(cloudProfileConfig.Constraints.FloatingPools, api.FloatingPool{
+				Name:                "*",
+				Region:              &region,
+				LoadBalancerClasses: lbClasses,
+			})
+
+			controlPlane.LoadBalancerClasses = []api.LoadBalancerClass{lbClasses[1]}
+
+			errorList := ValidateControlPlaneConfigAgainstCloudProfile(controlPlane, region, fpName, cloudProfile, cloudProfileConfig, nilPath)
 
 			Expect(errorList).To(BeEmpty())
 		})
 
+		It("should forbid for a load balancer class which is assigned by a wildcard FloatingPoolName because a more specific one exists", func() {
```

Comment:
Hm, not sure if I understand this case, can you elaborate?

Reply:
Sure.

Let's suppose you have this configuration in the cloud profile:

```
...
constraints:
  floatingPools:
  - name: fp-pool-1
    region: europe
    loadBalancerClasses:
    - name: lb-class-1
      floatingSubnetID: "1234"
      floatingNetworkID: "4567"
      subnetID: "7890"
  - name: "*"
    region: europe
    loadBalancerClasses:
    - name: lb-class-2
      floatingSubnetID: "1234"
      floatingNetworkID: "4567"
      subnetID: "7890"
```

If the shoot is configured for FP `fp-pool-1` it must not be allowed to let it use LB class `lb-class-2` because there is a closer match in the cloud profile that only specifies `lb-class-1`.

Let me revise the naming of this test but do you generally agree with this use case?

Reply:
Ah, understood now, yep, thanks!